spyder/preferences/languageserver.py

```diff
@@ -608,30 +607,356 @@ def mouseDoubleClickEvent(self, event):
+                code_style_codes),
+            'pycodestyle/select', alignment=Qt.Horizontal, word_wrap=False,
+            placeholder=_("Example codes: E113, W391"))
+        code_style_ignore = self.create_lineedit(
+            _("Ignore the following errors or warnings:"),
+            'pycodestyle/ignore', alignment=Qt.Horizontal, word_wrap=False,
+            placeholder=_("Example codes: E201, E303"))
+        code_style_max_line_length = self.create_spinbox(
+            _("Maximum allowed line length:"), None,
+            'pycodestyle/max_line_length', min_=10, max_=500, step=1,
+            tip=_("Default is 79"))
+
+        # Code style layout
+        code_style_g_layout = QGridLayout()
+        code_style_g_layout.addWidget(
+            self.code_style_filenames_match.label, 1, 0)
+        code_style_g_layout.addWidget(
+            self.code_style_filenames_match.textbox, 1, 1)
+        code_style_g_layout.addWidget(self.code_style_exclude.label, 2, 0)
+        code_style_g_layout.addWidget(self.code_style_exclude.textbox, 2, 1)
+        code_style_g_layout.addWidget(code_style_select.label, 3, 0)
+        code_style_g_layout.addWidget(code_style_select.textbox, 3, 1)
+        code_style_g_layout.addWidget(code_style_ignore.label, 4, 0)
+        code_style_g_layout.addWidget(code_style_ignore.textbox, 4, 1)
+        code_style_g_layout.addWidget(code_style_max_line_length.plabel, 5, 0)
+        code_style_g_layout.addWidget(
+            code_style_max_line_length.spinbox, 5, 1)
+
+        # Set Code style options enabled/disabled
+        code_style_g_widget = QWidget()
+        code_style_g_widget.setLayout(code_style_g_layout)
+        code_style_g_widget.setEnabled(self.get_option('pycodestyle'))
+        code_style_check.toggled.connect(code_style_g_widget.setEnabled)
+
+        # Code style layout
+        code_style_layout = QVBoxLayout()
+        code_style_layout.addWidget(code_style_label)
+        code_style_layout.addWidget(code_style_check)
+        code_style_layout.addWidget(code_style_g_widget)
+
+        code_style_widget = QWidget()
+        code_style_widget.setLayout(code_style_layout)
+
+        # --- Docstring tab ---
+        # Docstring style label
+        numpy_url = (
+            "<a href='https://numpydoc.readthedocs.io/en/"
+            "latest/format.html'>Numpy</a>")
+        pep257_url = (
+            "<a href='https://www.python.org/dev/peps/pep-0257/'>Pep 257</a>")
+        docstring_style_codes = _(
+            "<a href='http://www.pydocstyle.org/en/stable"
+            "/error_codes.html'>page</a>")
+        docstring_style_label = QLabel(
+            _("Here you can decide if you want to perform style analysis on "
+              "your docstrings according to the {} or {} conventions. You can "
+              "also decide if you want to show or ignore specific errors, "
+              "according to the codes found on this {}.").format(
```

Comment:
``Spyder can analyze your docstrings with pydocstyle for conformance to the basic {pep257_url} standard or the specific {numpydoc_url} convention. You can manually show or hide specific warnings by their <a href={pydocstyle_link}>pydocstyle error codes</a>.``

Reply:
TODO: (Later PR, if not simple) Add Googledoc, since we have Googledoc generation already?

Reply:
> TODO: (Later PR, if not simple) Add Googledoc, since we have Googledoc generation already?

Pydocstyle doesn't not provide such an option, just Numpy and Pep 257

Reply:
Thanks.